commit 699bb2e0c6f3796549dabac329501df7ffd99439
Author: Marc Zyngier <maz@kernel.org>
Date:   Sat May 28 12:38:18 2022 +0100

    KVM: arm64: Move vcpu PC/Exception flags to the input flag set
    
    The PC update flags (which also deal with exception injection)
    is one of the most complicated use of the flag we have. Make it
    more fool prof by:
    
    - moving it over to the new accessors and assign it to the
      input flag set
    
    - turn the combination of generic ELx flags with another flag
      indicating the target EL itself into an explicit set of
      flags for each EL and vector combination
    
    - add a new accessor to pend the exception
    
    This is otherwise a pretty straightformward conversion.
    
    Reviewed-by: Fuad Tabba <tabba@google.com>
    Reviewed-by: Reiji Watanabe <reijiw@google.com>
    Signed-off-by: Marc Zyngier <maz@kernel.org>

diff --git a/arch/arm64/kvm/hyp/nvhe/sys_regs.c b/arch/arm64/kvm/hyp/nvhe/sys_regs.c
index b6d86e423319..edd3eabf520f 100644
--- a/arch/arm64/kvm/hyp/nvhe/sys_regs.c
+++ b/arch/arm64/kvm/hyp/nvhe/sys_regs.c
@@ -38,9 +38,7 @@ static void inject_undef64(struct kvm_vcpu *vcpu)
 	*vcpu_pc(vcpu) = read_sysreg_el2(SYS_ELR);
 	*vcpu_cpsr(vcpu) = read_sysreg_el2(SYS_SPSR);
 
-	vcpu->arch.flags |= (KVM_ARM64_EXCEPT_AA64_EL1 |
-			     KVM_ARM64_EXCEPT_AA64_ELx_SYNC |
-			     KVM_ARM64_PENDING_EXCEPTION);
+	kvm_pend_exception(vcpu, EXCEPT_AA64_EL1_SYNC);
 
 	__kvm_adjust_pc(vcpu);